Dockerfile.development

```diff
@@ -4,42 +4,33 @@ FROM dependabot/dependabot-core
 USER root
 
 # Add debugging tools
-RUN apt-get update && apt-get install -y vim strace ltrace gdb
+RUN apt-get install -y vim strace ltrace gdb
 
-# This Dockerfile adds a non-root 'dependabot' user. However, for Linux,
-# this user's GID/UID must match your local user UID/GID to avoid permission issues
-# with bind mounts. Update USER_UID / USER_GID if yours is not 1000.
-# USER_UID should be the value of $UID from the host environment and
-# USER_GID the value of `id -g`.
-# See https://aka.ms/vscode-remote/containers/non-root-user for details.
-ARG USER_UID=1000
-ARG USER_GID=$USER_UID
-ARG USERNAME=dependabot
-
-RUN groupadd -o --gid "${USER_GID}" "${USERNAME}" && \
-  useradd --uid "${USER_UID}" --gid "${USER_GID}" -m "${USERNAME}"
-RUN chown -R "${USERNAME}":"${USERNAME}" \
+# FIXME: migrate ownership to earlier layers
+RUN chown -R dependabot:dependabot \
   /usr/local/.pyenv \
-  /opt/go/gopath \
-  /opt/rust/
-USER $USERNAME
-RUN git config --global user.name dependabot-ci && git config --global user.email no-reply@github.com
+  /opt/go/gopath
+
+USER dependabot
+RUN git config --global user.name dependabot-ci \
+  && git config --global user.email no-reply@github.com
 
-ARG CODE_DIR=/home/$USERNAME/dependabot-core
+ARG CODE_DIR=/home/dependabot/dependabot-core
 
 RUN curl -L -o ~/.vimrc https://github.com/hmarr/dotfiles/raw/main/vimrc-vanilla.vim && \
   echo 'export PS1="[dependabot-core-dev] \w \[$(tput setaf 4)\]$ \[$(tput sgr 0)\]"' >> ~/.bashrc
 
-RUN mkdir -p ${CODE_DIR}/common/lib/dependabot
-WORKDIR ${CODE_DIR}
-
-ENV BUNDLE_PATH="/home/$USERNAME/.bundle" \
+ENV BUNDLE_PATH="/home/dependabot/.bundle" \
   BUNDLE_BIN=".bundle/binstubs"
 ENV PATH="$BUNDLE_BIN:$PATH:$BUNDLE_PATH/bin"
 
-COPY common/Gemfile common/dependabot-common.gemspec ${CODE_DIR}/common/
-COPY common/lib/dependabot/version.rb ${CODE_DIR}/common/lib/dependabot/
-RUN cd common && bundle install
+COPY --chown=dependabot:dependabot common/Gemfile common/dependabot-common.gemspec ${CODE_DIR}/common/
+COPY --chown=dependabot:dependabot common/lib/dependabot/version.rb ${CODE_DIR}/common/lib/dependabot/
+
+WORKDIR ${CODE_DIR}
+RUN chown -R dependabot:dependabot $CODE_DIR
```

Comment:
Do we need to chown the code dir? Could we `mkdir` before copying as the `dependabot` user?